src/Compilers/Core/Portable/MetadataReference/AssemblyMetadata.cs

```diff
@@ -166,7 +168,7 @@ public static AssemblyMetadata CreateFromFile(string path)
 
         internal static AssemblyMetadata CreateFromFile(ModuleMetadata manifestModule, string path)
         {
-            return new AssemblyMetadata(manifestModule, moduleName => ModuleMetadata.CreateFromFile(Path.Combine(Path.GetDirectoryName(path), moduleName)));
+            return new AssemblyMetadata(manifestModule, moduleName => ModuleMetadata.CreateFromFile(Path.Combine(Path.GetDirectoryName(path) ?? "", moduleName)));
```

Comment:
>?? "" [](start = 141, length = 5)

Is this a bug fix?

Reply:
I think so.
`Path.Combine` throws when given a `null`, and I think it's possible to get `Path.GetDirectoryName(path)` to return `null` here by using a path in the root folder.